### PR TITLE
Introduce universal bootstrap method for tests with autoMinMasterNodes=false

### PR DESCRIPTION
**Pull request by andrershov**
Currently, there are a few tests that use autoMinMasterNodes=false and
hence override `addExtraClusterBootstrapSettings`, mostly this is 10-30
lines of codes that are copy-pasted from class to class.

This PR introduces `bootstrapMasterNodeWithSpecifiedId` which is
suitable for all classes and copy-paste could be removed. 

Removing code is always a good thing!

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java [hunk 1]
```diff
@@ -267,6 +268,17 @@
      */
     public static final String SYSPROP_THIRDPARTY = "tests.thirdparty";
 
+    /**
+     * See {@link #addExtraClusterBootstrapSettings(List)}.
+     * Please note that this value is being reset to -1 after each test. See {@link #resetBootstrapMasterNodeId()}.
+     */
+    protected int bootstrapMasterNodeId = -1;
```
Comment (DaveCTurner) [inline test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java]: Can we move this entirely within `InternalTestCluster`?

File: test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java [hunk 2]
```diff
@@ -1986,17 +1998,63 @@ public Settings transportClientSettings() {
         };
     }
 
+    /**
+     * Performs cluster bootstrap when {@link #bootstrapMasterNodeId} is started with the names of all
+     * previously started master-eligible nodes.
```
Comment (DaveCTurner) [inline test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java]: `s/previously started/existing and new/`?

File: test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java [hunk 3]
```diff
@@ -1986,17 +1998,63 @@ public Settings transportClientSettings() {
...
      * This method is called before starting a collection of nodes.
      * At this point the test has a holistic view on all nodes settings and might perform settings adjustments as needed.
      * For instance, the test could retrieve master node names and fill in
      * {@link org.elasticsearch.cluster.coordination.ClusterBootstrapService#INITIAL_MASTER_NODES_SETTING} setting.
+     * By default, this method delegates to {@link #bootstrapMasterNodeWithSpecifiedId(List)} method.
      *
      * @param allNodesSettings list of node settings before update
      * @return list of node settings after update
      */
     protected List<Settings> addExtraClusterBootstrapSettings(List<Settings> allNodesSettings) {
```
Comment (DaveCTurner) [inline test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java]: I think this is never overridden, so can be inlined.

**Review by DaveCTurner — COMMENTED** [review]
Great, the duplication irked me too.

I think it can be further simplified, moving this whole mechanism within `InternalTestCluster`. I also wonder if we should reset `bootstrapMasterNodeId` to `-1` after use, and should assert that it is `-1` if `autoManageMinMasterNodes` is set.

Also there's a few places where you org.elasticsearch.use org.elasticsearch.fully-qualified org.elasticsearch.names, mirroring existing usage. I think we can tidy that up, at least in the places they occur here. There's only one `NODE_NAME_SETTING` worth talking about.

**Comment by andrershov** [comment]
@DaveCTurner thanks for your review!
And thank you for the suggestion to move bootstrap logic to `InternalTestCluster`, after this change `addExtraClusterBootrstrap` method could be removed from `NodeConfigurationSource` and `ESIntegTestCase`. If in the future, we will require a method to add some settings before nodes are started, it can easily be returned back.
I've made all the requested changes and I think it makes sense to have the second review round.

**Comment by andrershov** [comment]
I've missed that `InternalTestClusterTests` also overrides `addExtraClusterBootstrapSettings`. Now it can happily go away.

**Review by DaveCTurner — APPROVED** [review]
LGTM, nice work

**Pull request merged**